frontend/src/pages/single-booking.tsx

```diff
@@ -0,0 +1,161 @@
+  const handleNavigation = (buttonValue: string) => {
+    if (buttonValue === "go-to-input-student-data") {
+      setShowInputBikeSize(false);
+      setShowInputStudentData(true);
+      setShowBookingPreConfirmation(false);
+    } else if (
+      buttonValue === "go-to-pre-booking-confirmation" &&
+      isUserDataValid
+    ) {
+      setShowInputBikeSize(false);
+      setShowInputStudentData(false);
+      setShowBookingPreConfirmation(true);
+    } else if (buttonValue === "return-to-size-selection") {
+      setShowInputBikeSize(true);
+      setShowInputStudentData(false);
+      setShowBookingPreConfirmation(false);
+    } else if (buttonValue === "return-to-user-data-input") {
+      setShowInputBikeSize(false);
+      setShowInputStudentData(true);
+      setShowBookingPreConfirmation(false);
+    }
+  };
+
+  // Update bikeSizeSelected
+  function handleBikeSize(event: { selectedSize: BikeSize }) {
+    const bikeSizeSelected = event.selectedSize;
+    setBikeSize(bikeSizeSelected);
+  }
+
+  // Correct me
+  // Create a function to validate user data input
+  function checkEnteredUserData() {
+    setIsUserDataValid(true);
+  }
+
+  // Update Booking data after both states change
+  useEffect(() => {
+    setBookingData((prevBookingData) => ({
+      ...prevBookingData,
+      bookingUserData: enteredUserData,
+    }));
+    setBookingData({
+      bookingBikeSize: bikeSizeSelected,
+      bookingUserData: enteredUserData,
+    });
+    checkEnteredUserData();
+  }, [bikeSizeSelected, isUserDataValid, enteredUserData]);
+
+  // Option to confirm Booking or Return to user data input
+  function handleBookingConfirmation(event: { name: any }) {
+    const buttonOnConfirmation = event.name;
+    if (buttonOnConfirmation === "submit-booking") {
+      alert(
+        buttonOnConfirmation +
+          "\n" +
+          "Bike Size Selected : " +
+          bookingData.bookingBikeSize +
+          "\n" +
+          "User Name: " +
+          bookingData.bookingUserData.firstName +
+          " " +
+          bookingData.bookingUserData.lastName +
+          "\n" +
+          "User Room: " +
+          bookingData.bookingUserData.roomNumber +
+          "\n"
+      );
+      //function to submit data
+    } else if (buttonOnConfirmation === "return-to-user-data-input") {
+      setIsUserDataValid(false);
+    }
+  }
+
+  // Correct me (remove)
+  // Checking status of bikeSizeSelected (bike size input) and isUserDataValid (user data input)
+  useEffect(() => {
+    console.log("bikeSizeSelected changed:", bikeSizeSelected);
+  }, [bikeSizeSelected]);
+  useEffect(() => {
+    console.log("isUserDataValid changed:", isUserDataValid);
+  }, [isUserDataValid]);
+  useEffect(() => {
+    console.log("enteredUserData changed:", enteredUserData);
+  }, [enteredUserData]);
+
+  return (
+    <div className="container center-content">
+      <HeaderTemp heading="Single Booking" />
+
+      {showInputBikeSize && (
```

Comment:
Could we change these booleans for a more compact way of representing the page state, like a Enum? 

Reply:
I made a new version following this idea. Thank you.